docs_guidelines/how_to/add_code.rst

```diff
@@ -0,0 +1,149 @@
+
+.. code-block:: text
+
+    the output of ``2>3`` is ``False``
+
+Code blocks
+===========
+
+There are several ways to include code cells in your documentation:
+
+* Executable cells with ``jupyter-execute``.
+* Non-executable cells with ``code-block``.
+* Testable code cells with ``doctest``.
+
+``jupyter-execute``
+-------------------
+
+If you want to run Python code that includes visualization, you can use ``jupyter-execute`` to include cells that are executed in a Jupyter kernel and show the output of that code in your documentation. The syntax is:
+
+
+.. code-block:: text
+
+    .. jupyter-execute::
+
+        your_code
+
+For example, you can write this:
+
+.. code-block:: text
+
+    .. jupyter-execute::
+
+        from qiskit import QuantumCircuit
+
+        qc = QuantumCircuit(2)
+        qc.h(0)
+        qc.cx(0,1)
+        qc.draw('mpl')
+
+The output would be this cell:
+
+.. jupyter-execute::
+
+    from qiskit import QuantumCircuit
+
+    qc = QuantumCircuit(2)
+    qc.h(0)
+    qc.cx(0,1)
+    qc.draw('mpl')
+
+
+.. note::
+
+    Even though ``jupyter-execute`` can be used for cells without any visualization, it makes the documentation building process
+    more complex, so it is recommended to use ``doctest`` instead. See `this page <https://github.com/Qiskit/qiskit-terra/issues/7661>`_ for more details.
+
+.. note::
+
+    For the ``jupyter-execute`` cells to appear you need to add the extension ``jupyter_sphinx`` to the ``docs/conf.py`` of your repository.
+
+
+
+``code-block``
+--------------
+
+There are some situations in which executing the code is not convenient, such as when:
+
+* The code you are showing is not written in Python but other languages like reStructuredText, YAML, Markdown or Rust.
+* The code requires connecting to a provider.
+* The code takes too long to run.
+
+In those cases, you can use ``code-block``, whose syntax is:
+
+.. code-block:: text
+
+    .. code-block:: language
+
+        your_code
+
+You can pick as ``language`` any of the short names of the lexers supported by `Pygments <https://pygments.org/docs/lexers/#>`_, like ``python``, ``bash`` or ``text``.
+For example, you can write this:
+
+.. code-block:: text
+
+    .. code-block:: bash
+
+        pip install qiskit
+
+
+And the output will look like this:
+
+.. code-block:: bash
+
+    pip install qiskit
+
+
+``doctest``
+-----------
+
+If you want to write Python code cells that don't include visualizations and check if they work as intended, you can use ``doctest``, whose syntax is:
+
+.. code-block:: text
+
+    .. doctest::
+
+        >>> your_code
+        expected_output
+
+That way, ``doctest`` runs ``your_code`` and checks whether the output is ``expected_output``.
+As an example, you can write this:
+
+.. code-block:: text
+
+    .. doctest::
+
+        >>> print(3+2)
+        5
+
+Then this cell would be run:
+
+.. doctest::
+
+    >>> print(3+2)
+    5
```

Comment:
I tried changing the expected output to other values like `6` but it didn't give me any error. Can you investigate on that?

Reply:
To actually run the tests you need to run ``sphinx-build -b doctest source_dir output_dir``. Added explanation and example of the output.